indexing-service/src/main/java/org/apache/druid/indexing/overlord/config/RemoteTaskRunnerConfig.java

```diff
@@ -81,15 +83,14 @@ public Period getTaskCleanupTimeout()
 
   public int getMaxZnodeBytes()
   {
-    return maxZnodeBytes;
+    return (int) maxZnodeBytes.getValue();
```

Comment:
How about adding `asInt()` which has a check for integer overflow?

Reply:
A new `getBytesInInt` is provided to check the potential overflow.